.github/workflows/java-cd.yml

```
@@ -158,3 +143,112 @@ jobs:
                   name: java-${{ matrix.host.TARGET }}
                   path: |
                       java/bundle*.jar
+
+    publish-to-maven-central-deployment:
+        if: ${{ inputs.maven_publish == true || github.event_name == 'push' }}
+        needs: [set-release-version, create-binaries-to-publish]
+        runs-on: ubuntu-latest
+        outputs:
+            DEPLOYMENT_ID: ${{ steps.maven-deployment.outputs.DEPLOYMENT_ID }}
+        env:
+            RELEASE_VERSION: ${{ needs.set-release-version.outputs.RELEASE_VERSION }}
+        steps:
+            - name: Download published artifacts
+              uses: actions/download-artifact@v4
+
+            - name: Move all required files to one directory
+              run: |
+                mkdir maven-files
+                cd maven-files
+                for file in $(find ../. -name "*.jar"); do jar xf "$file" ; done
+
+            - name: Generate sha1 and md5 files for all Maven files
+              run: |
+                cd maven-files
+                for i in *.jar *.pom *.module; do md5sum $i | cut -d ' ' -f 1 > $i.md5; done
+                for i in *.jar *.pom *.module; do sha1sum $i | cut -d ' ' -f 1 > $i.sha1; done
+
+            - name: Move files to the correct directory tree
+              run: |
+                  mkdir -p build/io/valkey/valkey-glide/${{ env.RELEASE_VERSION }}
+                  cp -a maven-files/* build/io/valkey/valkey-glide/${{ env.RELEASE_VERSION }}
+                  rm -rf build/io/valkey/valkey-glide/${{ env.RELEASE_VERSION }}/META-INF
+                  cd build
+                  zip -r ../build io
+
+            - name: Upload artifacts to publish
+              uses: actions/upload-artifact@v4
+              with:
+                name: valkey-${{ env.RELEASE_VERSION }}
+                path: |
+                  build
+
+            - name: Publish to Maven Central
+              id: maven-deployment
+              run: |
+                  BEARER_TOKEN=$(echo "${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}" | base64)
+                  export DEPLOYMENT_ID=`curl --request POST \
+                  --header "Authorization: Bearer $BEARER_TOKEN" \
+                  --form bundle=@build.zip \
+                  https://central.sonatype.com/api/v1/publisher/upload | tail -n 1`
+                  echo "DEPLOYMENT_ID=$DEPLOYMENT_ID" >> $GITHUB_ENV
+                  echo "BEARER_TOKEN=$BEARER_TOKEN" >> $GITHUB_ENV
+                  echo "DEPLOYMENT_ID=$DEPLOYMENT_ID" >> $GITHUB_OUTPUT
+                  echo $DEPLOYMENT_ID
+
+            - name: Check status of deployment
+              run: |
+                  sleep 20
+                  export DEPLOYMENT_STATUS=`curl --request POST \
+                  --header "Authorization: Bearer ${{ env.BEARER_TOKEN }}" \
```

```suggestion
                  -u "${{ secrets.CENTRAL_TOKEN_USERNAME }}:${{ secrets.CENTRAL_TOKEN_PASSWORD }}" \
```
Update it in all places